File: solutions/D. 196(small).py
import sys
input = sys.stdin.readline

PLUS = 500 #기본적으로 인덱스 값

T = int(input())
for Case in range(1,T+1):
    N=int(input())
    Numbers = [0]*(1003) #쌍들을 배열로 저장
    cnt=0
    for num in input().split():
        num = int(num)
        cnt+=Numbers[196-num + PLUS]
        Numbers[num+PLUS]+=1
    print(f'Case #{Case} : {cnt}')

File: solutions/D. 196(small).cpp
#include <iostream>
#define fast ios::sync_with_stdio(0),cin.tie(0)
#define endl '\n'
/*
    pb : D. 196(small)
*/
const int PLUS = 500; //인덱스 값
using namespace std;
int main(){
    fast;
    int T;cin >> T;
    for(int Case=1;Case<=T;Case++){
        int N;cin >> N;
        int Number[1300]{},cnt=0;
        while(N--){
            int num;cin >> num;
            cnt+=Number[196-num+PLUS];
            Number[num+PLUS]++;
        }
        cout << "Case #" << Case << " : " << cnt << endl;
    }
    return 0;
}